pkg/types/relayer.go

```diff
@@ -78,33 +78,38 @@ type NodeStatus struct {
 	State   string
 }
 
-// ChainService is a sub-interface that encapsulates the explicit interactions with a chain, rather than through a provider.
-type ChainService interface {
-	Service
-
+type ChainReader interface {
+	// NewContractReader returns a new ContractReader.
+	// The format of contractReaderConfig depends on the implementation.
+	NewContractReader(ctx context.Context, contractReaderConfig []byte) (ContractReader, error)
 	// LatestHead returns the latest head for the underlying chain.
 	LatestHead(ctx context.Context) (Head, error)
 	// GetChainStatus returns the ChainStatus for this Relayer.
 	GetChainStatus(ctx context.Context) (ChainStatus, error)
 	// ListNodeStatuses returns the status of RPC nodes.
 	ListNodeStatuses(ctx context.Context, pageSize int32, pageToken string) (stats []NodeStatus, nextPageToken string, total int, err error)
+}
+
+type IChainWriter interface {
```

Comment:
This is a bummer... :thinking: We could rename it later of course - are you planning to do that? Or leave the deprecated type around for a while?

Reply:
Everyone that has context on this is slammed with work, I've been chipping away at the backlog when I have time. Maybe I'll do it, or I'll try to somehow prioritise the backlog

Reply:
Is this conflict resolved now?
```suggestion
type ChainWriter interface {
```